src/pages/Map.js

```diff
@@ -1,50 +1,70 @@
-import React, { useState } from 'react';
+import React, { useMemo, useState, useEffect, useCallback } from 'react';
+import { useSearchParams, useNavigate } from 'react-router-dom';
+
 import PhilippinesMapJSX from '../PhilippinesMapJSX';
-import ".././App.css";
+import { provinces, menuOptions } from '../utils/province';
+import '.././App.css';
+import {
+  levelArrayToString,
+  levelStringToArray,
+} from '../utils/levelConverter';
 
 const PhilippinesMap = () => {
-  const [provinceLevels] = useState([]);
-  const [selectedProvince, setSelectedProvince] = useState(null);
-  const [selectedProvinceLayer, setSelectedProvinceLayer] = useState(null);
+  const navigate = useNavigate();
+  const [searchParams] = useSearchParams();
+
+  const [provinceLevels, setProvinceLevels] = useState(new Array(83).fill(0));
```

Comment:
Generalize the length of the array, it shouldn't be fixed to a certain length. I would go for something like taking the length of the actual list of polygons per map to be safe.